luamocks/autogenerated/VBO.lua

```diff
@@ -0,0 +1,163 @@
+---VBO
+
+---
+---@return nil
+function vbo:Delete() end
+
+---Parameters
+---@param size number
+---@param attribs number|{{number,number,number,number,number},...}
+---@return nil
+function vbo:Define(size, attribs) end
+
+---terrainVertexVBO:Define(numPoints, { {id = 0, name = "pos", size = 2}, })
+---@return number elementsCount
+function vbo:GetBufferSize() end
+
+---@return number bufferSizeInBytes
+function vbo:GetBufferSize() end
+
+---@return number size
+function vbo:GetBufferSize() end
+
+---Parameters
+---@param vboData {number,...}
+---@param attributeIndex number (default): `-1`
+---@param elemOffset number (default): `0`
+---@param luaStartIndex number (default): `0`
+---@param luaFinishIndex number (optional)
+---@return {number, ...} indexData
+function vbo:Upload(vboData[, attributeIndex=-1[, elemOffset=0[, luaStartIndex=0[, luaFinishIndex]]]]) end
+
+---@return number elemOffset
+function vbo:Upload(vboData[, attributeIndex=-1[, elemOffset=0[, luaStartIndex=0[, luaFinishIndex]]]]) end
+
+---@return number|{number,number,number,number} attrID
+function vbo:Upload(vboData[, attributeIndex=-1[, elemOffset=0[, luaStartIndex=0[, luaFinishIndex]]]]) end
+
+---vbo:Upload(posArray, 0, 1)
+-- 0 is offset into vbo (on GPU) in this case no offset
+-- 1 is lua index index into the Lua table, in this case it's same as default
+-- Upload will upload from luaOffset to end of lua array
+---rectInstanceVBO:Upload({1},0)
+---Parameters
+---@param attributeIndex number (default): `-1`
+---@param elementOffset number (default): `0`
+---@param elementCount number (optional)
+---@param forceGPURead boolean (default): `false`
+---@return {{number,...},...} vboData
+function vbo:Download([attributeIndex=-1[, elementOffset=0[, elementCount[, forceGPURead=false]]]]) end
+
+---@return nil|number buffer size in bytes
+function vbo:ModelsVBO() end
+
+---Parameters
+---@param unitDefIDs number|{number,...}
+---@param attrID number
+---@param teamIdOpt number (optional)
+---@param elementOffset number (optional)
+---@return {number,number,number,number} instanceData
+function vbo:InstanceDataFromUnitDefIDs(unitDefIDs, attrID[, teamIdOpt[, elementOffset]]) end
+
+---@return number elementOffset
+function vbo:InstanceDataFromUnitDefIDs(unitDefIDs, attrID[, teamIdOpt[, elementOffset]]) end
+
+---@return attrID
+function vbo:InstanceDataFromUnitDefIDs(unitDefIDs, attrID[, teamIdOpt[, elementOffset]]) end
+
+---Data Layout
+
+SInstanceData:
+   , matOffset{ matOffset_ }            // updated during the following draw frames
+   , uniOffset{ uniOffset_ }            // updated during the following draw frames
+   , info{ teamIndex, drawFlags, 0, 0 } // not updated during the following draw frames
+   , aux1 { 0u }
```

Comment:
doesn't look like valid lua, will that be a problem?

Reply:
It is not necessarily a problem but needs to be removed, which I am doing now.